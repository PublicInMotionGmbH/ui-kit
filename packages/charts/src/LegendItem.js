import React from 'react'
import PropTypes from 'prop-types'

import { buildClassName } from '@talixo/shared'

import { getColorIndex } from './utils'

export const moduleName = 'legend-item'

const propTypes = {
  /** Additional wrapper className */
  className: PropTypes.string,

  /** Colorbox color */
  color: PropTypes.string,

  /** Indicates if item is disabled */
  disabled: PropTypes.bool,

  /** Id of the element */

  id: PropTypes.oneOfType([ PropTypes.string, PropTypes.number ]),

  /** onClick callback function */
  onClick: PropTypes.func,

  /** Data series name */
  label: PropTypes.string,

  /** Data series value */
  value: PropTypes.oneOfType([ PropTypes.string, PropTypes.number ])
}

const defaultProps = {
  disabled: false
}

/**
 *
 * @param {object} [props]
 * @param {string} [props.className]
 * @param {string} [props.color]
 * @param {bool} [props.disabled]
 * @param {object} [props.id]
 * @param {function} [props.onClick]
 * @param {string} [props.label]
 * @param {string|number} [props.value]
 * @returns {ReactElement}
 */
function LegendItem (props) {
  const { className, color, disabled, id, onClick, label, value } = props
  const itemCls = buildClassName(`${moduleName}`, className, {
    clickable: onClick,
    disabled: disabled
  })
  const colorboxCls = buildClassName(`${moduleName}__colorbox`, null, [getColorIndex(id)])
  const labelCls = buildClassName(`${moduleName}__label`)

  return (
    <div onClick={onClick} className={itemCls}>
      <div
        style={(color && { backgroundColor: color }) || {}}
        className={colorboxCls}
      />
      <div className={labelCls}>{ label }</div>
      <div className={labelCls}> { value && `(${value})` }</div>
    </div>
  )
}

LegendItem.propTypes = propTypes

LegendItem.defaultProps = defaultProps

LegendItem.displayName = 'LegendItem'

export default LegendItem
