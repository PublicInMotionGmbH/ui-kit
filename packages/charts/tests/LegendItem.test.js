import React from 'react'
import LegendItem from '../src/LegendItem'
import { shallow } from 'enzyme'

const createProps = (props = {}) => ({
  ...props
})
const defaultProps = createProps()
const createWrapper = (props = defaultProps) => shallow(<LegendItem {...props} />)

describe('<LegendItem> component ', () => {
  let wrapper
  describe('rendering', () => {
    beforeEach(() => {
      wrapper = createWrapper()
    })
    it('should render properly', () => {
      expect(wrapper).toMatchSnapshot()
    })
  })
})
