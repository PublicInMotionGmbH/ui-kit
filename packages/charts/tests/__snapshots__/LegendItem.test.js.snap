// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LegendItem> component  rendering should render properly 1`] = `
<div
  className="talixo-legend-item test talixo-legend-item--clickable"
  onClick={[Function]}
>
  <div
    className="talixo-legend-item__colorbox talixo-legend-item__colorbox--0"
    style={
      Object {
        "backgroundColor": "white",
      }
    }
  />
  <div
    className="talixo-legend-item__title"
  >
    Title
     
    (10)
  </div>
</div>
`;
